gate-web/src/main/groovy/com/netflix/spinnaker/gate/services/ApplicationService.groovy

```diff
@@ -79,23 +79,37 @@ class ApplicationService {
     })
   }
 
-  void tick() {
-    def applicationListRetrievers = buildApplicationListRetrievers()
+  /**
+   * Fetching cluster details is a potentially expensive call to cloud driver, but allows us to provide a definitive
+   * list of accounts that an application has a presence in.
+   *
+   * As a trade-off, we'll fetch cluster details on the background refresh loop and merge in the account details
+   * when applications are requested on-demand.
+   *
+   * @param expandClusterNames Should cluster details (for each application) be fetched from cloud driver
+   * @return Applications
+   */
+  List<Map<String, Object>> tick(boolean expandClusterNames = true) {
+    def applicationListRetrievers = buildApplicationListRetrievers(expandClusterNames)
     List<Future<List<Map>>> futures = executorService.invokeAll(applicationListRetrievers)
     List<List<Map>> all = futures.collect { it.get() }
     List<Map> flat = (List<Map>) all?.flatten()?.toList()
-    def mergedApplications = mergeApps(flat, serviceConfiguration.getService('front50')).collect {
+    return mergeApps(flat, serviceConfiguration.getService('front50')).collect {
       it.attributes
     } as List<Map>
-
-    allApplicationsCache.set(mergedApplications)
   }
 
   List<Map> getAll() {
     try {
-      tick()
-    } catch (ignored) {
-      // do nothing
+      def applicationsByName = allApplicationsCache.get().groupBy { it.name }
+      return tick(false).collect { Map application ->
+        applicationsByName[application.name.toString()]?.each { Map cacheApplication ->
+          application.accounts = mergeAccounts(application.accounts as String, cacheApplication.accounts as String)
```

Comment:
Merge in the cached application account information (which comes from Front50 + Cloud Driver) w/ the current application account information (which only comes from Front50).

The refresh happens every 60s so there's a possibility of a _slight_ delay ... but unlikely to bother anyone as we'll refresh prior to the rest of the system even realizing a new server group has been deployed.

This does raise the question as to whether we should be making a decision around a single source of truth for application account metadata rather than letting it be both user defined and system derived?